library/std/src/sys/os_str/bytes.rs

```diff
@@ -216,19 +216,26 @@ impl Buf {
         self.as_slice().into_rc()
     }
 
-    /// Provides plumbing to core `Vec::truncate`.
-    /// More well behaving alternative to allowing outer types
-    /// full mutable access to the core `Vec`.
-    #[inline]
-    pub(crate) fn truncate(&mut self, len: usize) {
+    /// Provides plumbing to `Vec::truncate` without giving full mutable access
+    /// to the `Vec`.
+    ///
+    /// # Safety
+    ///
+    /// The length must be at an `OsStr` boundary, according to
+    /// `Slice::check_public_boundary`.
```

Comment:
I am having increasingly negative opinions about having an internal type named `Slice` but it is not your responsibility for that, so

Reply:
Then what would a better name be? I might submit a PR for that. But maybe not.

Reply:
I am not sure, honestly! it just is starting to bug me because this looked like a typo at first.

Reply:
See #140936 for a change to the type name